Testshell/main.cpp

```diff
@@ -139,13 +142,138 @@ TEST_F(TestShellFixture, ReadLbaOverThanMaxOfInt) {
   int MAX_INT = 2147483647; // Maximum value for a 32-bit signed integer
   Command command{"read", vector<string>{"2147483648"}};
 
-  EXPECT_CALL(ssd, read(_)).Times(0);
+  EXPECT_CALL(ssd, read(_))
+    .Times(0);
 
-  EXPECT_CALL(ssd, getResult()).Times(0);
+  EXPECT_CALL(ssd, getResult())
+    .Times(0);
 
   ts.executeCommand(command);
 }
 
+TEST_F(TestShellFixture, FullReadNormalCase) {
+  vector<string> args{};
+  // Total LBA 수를 3인 경우로 가정
+  // 3개의 LBA를 성공적으로 읽고 4번째에서 ERROR 반환
+  EXPECT_CALL(ssd, read(0))
+    .Times(1);
+  EXPECT_CALL(ssd, read(1))
+    .Times(1);
+  EXPECT_CALL(ssd, read(2))
+    .Times(1);
+  EXPECT_CALL(ssd, read(3))
+    .Times(1);
+
+  EXPECT_CALL(ssd, getResult())
+    .Times(4)
+    .WillOnce(Return("0xAAAABBBB"))
+    .WillOnce(Return("0xCCCCDDDD"))
+    .WillOnce(Return("0xEEEEFFFF"))
+    .WillOnce(Return("ERROR"));
+
+  CaptureStdout();
+  ts.fullread(args);
+  std::string output = GetCapturedStdout();
+
+  EXPECT_TRUE(output.find("[Full Read] LBA: 0 Result: 0xAAAABBBB") !=
+              std::string::npos);
+  EXPECT_TRUE(output.find("[Full Read] LBA: 1 Result: 0xCCCCDDDD") !=
+              std::string::npos);
+  EXPECT_TRUE(output.find("[Full Read] LBA: 2 Result: 0xEEEEFFFF") !=
+              std::string::npos);
+}
+
+TEST_F(TestShellFixture, FullReadInvalidUsage) {
+  vector<string> args{"extra"};
+  EXPECT_CALL(ssd, read(_))
+    .Times(0);
+  EXPECT_CALL(ssd, getResult())
+    .Times(0);
+
+  CaptureStdout();
+  ts.fullread(args);
+  std::string output = GetCapturedStdout();
+  EXPECT_EQ("INVALID COMMAND\n", output);
+}
+
+// fullwrite 테스트들
+TEST_F(TestShellFixture, FullWriteNormalCase) {
+  vector<string> args{"42"};
+
+  // 3개의 LBA에 성공적으로 쓰고 4번째에서 ERROR 반환
+  EXPECT_CALL(ssd, write(0, 42))
+    .Times(1);
+  EXPECT_CALL(ssd, write(1, 42))
+    .Times(1);
+  EXPECT_CALL(ssd, write(2, 42))
+    .Times(1);
+  EXPECT_CALL(ssd, write(3, 42))
+    .Times(1);
+
+  EXPECT_CALL(ssd, getResult())
+    .Times(4)
+    .WillOnce(Return("SUCCESS"))
+    .WillOnce(Return("SUCCESS"))
+    .WillOnce(Return("SUCCESS"))
+    .WillOnce(Return("ERROR"));
+
+  CaptureStdout();
+  ts.fullwrite(args);
+  std::string output = GetCapturedStdout();
+
+  EXPECT_TRUE(output.find("[Full Write] LBA: 0 Value: 42") !=
+              std::string::npos);
+  EXPECT_TRUE(output.find("[Full Write] LBA: 1 Value: 42") !=
+              std::string::npos);
+  EXPECT_TRUE(output.find("[Full Write] LBA: 2 Value: 42") !=
+              std::string::npos);
+}
```

Comment:
마지막 LBA 가 출력되지 않도록  UT에 추가하는 거 어떤가요?

```suggestion
  EXPECT_TRUE(output.find("[Full Write] LBA: 2 Value: 42") !=
              std::string::npos);
  EXPECT_TRUE(output.find("[Full Write] LBA: 3") ==
              std::string::npos);
}
```

Reply:
좋은 테스트인 것 같습니다! 추가 하겠습니다. 감사합니다.